### PR TITLE
feat(glean): Add 2FA 'these authentication apps' link click event, adjust DataBlock data attrs

### PR DESCRIPTION
**Pull request by LZoog**
Because:
* We want metrics for when a user clicks on this link
* Simplifies glean data attrs in DataBlock because the type is string or undefined, and if this is undefined the attribute does not render

This commit:
* Adds the glean attrs on LinkExternal
* Shares 2fa glean data 'type's for consistency

closes FXA-9578

### CODE REVIEW

File: packages/fxa-react/components/LinkExternal/index.tsx
Comment (chenba) [inline packages/fxa-react/components/LinkExternal/index.tsx:17]: Seems strange to me that a 2fa specific type is used in a much more generic type for link props.

File: packages/fxa-react/lib/interfaces.ts
Comment (chenba) [inline packages/fxa-react/lib/interfaces.ts:5]: I'm not afraid of spaces, but I also don't know how Glean is going to handle them.

**Review by chenba — APPROVED** [review]
👍  but I think the type for the Glean "type" in LinkExternal should be generic.

File: packages/fxa-react/components/LinkExternal/index.tsx
Comment (LZoog) [reply packages/fxa-react/components/LinkExternal/index.tsx:17]: Sure, but wouldn't we just expand it at that time? That's what I pictured but, I can change to 'string'. (Maybe this should be an enum anyways...)

File: packages/fxa-react/lib/interfaces.ts
Comment (LZoog) [reply packages/fxa-react/lib/interfaces.ts:5]: Hmmm, well I'm certainly open to using a dash or something instead.

File: packages/fxa-react/components/LinkExternal/index.tsx
Comment (chenba) [reply packages/fxa-react/components/LinkExternal/index.tsx:17]: > but wouldn't we just expand it at that time?

Does it make sense to have 'inline setup' always as a valid value of that type when applied to LinkExternal?  I'm arguing no and we might as well be generic.  I could see using a type argument to limit the valid values to whatever context in which the LinkExternal is used.

Comment (LZoog) [reply packages/fxa-react/components/LinkExternal/index.tsx:17]: Eh yeah good point. Mostly I just wanted us to not typo the `type` here but I'll wait for #17459 and #17460 to land and then I'll adjust this and turn that 2fa type into an enum and update all around, will likely request re-review.

**Comment by LZoog** [comment]
Just for posterity: this was r+ but I pushed a small update addressing a comment and added that enum for 2fa Glean attr 'type's and updated the references which was discussed in Slack - since another issue depends on this one for the `LinkExternal` updates, I'll merge once CI goes green.

**Pull request merged**
